src/internal/ppsutil/decoder_test.go

```diff
@@ -0,0 +1,114 @@
+package ppsutil_test
+
+import (
+	"encoding/json"
+	"io"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/ppsutil"
+)
+
+func TestSpecReader(t *testing.T) {
+	var cases = map[string]struct {
```

Comment:
I usually put a `name` field in here: https://github.com/golang/go/wiki/TestComments#choose-human-readable-subtest-names

Reply:
That’s what the map key serves to do. I don’t think order matters here, hence a map vice a slice.